Tests/OpenTelemetryInstrumentationSupportTests/SpanAttributeOpenTelemetryTests.swift

```diff
@@ -0,0 +1,98 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Tracing open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Tracing project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Baggage
+import Instrumentation
+import NIOHTTP1
+import OpenTelemetryInstrumentationSupport
+import TracingInstrumentation
+import XCTest
+
+final class SpanAttributeOpenTelemetryTests: XCTestCase {
+    func testSemanticAttributes() {
+        InstrumentationSystem.bootstrap(FakeTracer())
+        var span = InstrumentationSystem.tracingInstrument.startSpan(named: "test", context: BaggageContext())
+
+        span.setAttribute(418, forKey: SpanAttribute.HTTP.StatusCode.self)
```

Comment:
so the typed keys are great.

I'd like to see if we can improve two things here:
1) if we can figure out a way to write these as `.HTTP.statusCode`
2) avoiding the `.self` there, it's a bit annoying.

2 we can solve by:

```swift

protocol AttrKey {
    associatedtype Value
}

enum Attr: AttrKey {
    typealias Value = Never
    enum HTTP {
        static let statusCode = StatusCode()
        struct StatusCode: AttrKey {
            typealias Value = Int
        }
    }
}

func take<K>(_ v: K.Value, forKey key: K) where K: AttrKey {}

func test() {
    take(200, forKey: Attr.HTTP.statusCode)
    take("nein", forKey: Attr.HTTP.statusCode)
}

```

for 1 I'm not sure...